ggml-vulkan.cpp

```diff
@@ -1745,39 +1745,45 @@ void ggml_vk_instance_init() {
 
         // Default to using all dedicated GPUs
         for (size_t i = 0; i < devices.size(); i++) {
-            vk::PhysicalDeviceProperties props = devices[i].getProperties();
-
-            if (props.deviceType == vk::PhysicalDeviceType::eDiscreteGpu) {
+            vk::PhysicalDeviceProperties new_props_1 = devices[i].getProperties();
```

Comment:
This isn't needed, you can get the PhysicalDeviceProperties from `new_props_2.properties`. Then you also don't need to call it "new_props_2`, just `new_props`